magma/backend/mlir/hardware_module.py

```diff
@@ -365,6 +402,17 @@ def visit_primitive(self, module: ModuleWrapper) -> bool:
 
     @wrap_with_not_implemented_error
     def visit_magma_mux(self, module: ModuleWrapper) -> bool:
+
+        def make_mux(data, select, result):
+            mlir_type = hw.ArrayType((len(data),), data[0].type)
+            array = self._ctx.new_value(mlir_type)
+            hw.ArrayCreateOp(
+                operands=data,
+                results=[array])
+            hw.ArrayGetOp(
+                operands=[array, select],
+                results=[result])
+
```

Comment:
Should we move this out of the enclosing function def since it doesn't use any references to the enclosing scope?